src/nbl/video/SOpenGLContextLocalCache.h

```diff
@@ -0,0 +1,149 @@
+            constexpr size_t toFill = IGPUMeshBuffer::MAX_PUSH_CONSTANT_BYTESIZE / sizeof(uint64_t);
+            constexpr size_t bytesLeft = IGPUMeshBuffer::MAX_PUSH_CONSTANT_BYTESIZE - (toFill * sizeof(uint64_t));
+            constexpr uint64_t pattern = 0xdeadbeefDEADBEEFull;
+            std::fill(reinterpret_cast<uint64_t*>(pushConstantsState<PBP>()->data), reinterpret_cast<uint64_t*>(pushConstantsState<PBP>()->data) + toFill, pattern);
+            if constexpr (bytesLeft > 0ull)
+                memcpy(reinterpret_cast<uint64_t*>(pushConstantsState<PBP>()->data) + toFill, &pattern, bytesLeft);
+            //#endif
+
+            _stages |= IGPUSpecializedShader::ESS_ALL;
+        }
+        pushConstantsState<PBP>()->incrementStamps(_stages);
+
+        pushConstantsState<PBP>()->layout = core::smart_refctd_ptr<const COpenGLPipelineLayout>(_layout);
+        memcpy(pushConstantsState<PBP>()->data + _offset, _values, _size);
+    }
+
+private:
+    static inline GLenum getGLpolygonMode(asset::E_POLYGON_MODE pm)
+    {
+        const static GLenum glpm[3]{ GL_FILL, GL_LINE, GL_POINT };
+        return glpm[pm];
+    }
+    static inline GLenum getGLcullFace(asset::E_FACE_CULL_MODE cf)
+    {
+        const static GLenum glcf[4]{ 0, GL_FRONT, GL_BACK, GL_FRONT_AND_BACK };
+        return glcf[cf];
+    }
+    static inline GLenum getGLstencilOp(asset::E_STENCIL_OP so)
+    {
+        static const GLenum glso[]{ GL_KEEP, GL_ZERO, GL_REPLACE, GL_INCR, GL_DECR, GL_INVERT, GL_INCR_WRAP, GL_DECR_WRAP };
+        return glso[so];
+    }
+    static inline GLenum getGLcmpFunc(asset::E_COMPARE_OP sf)
+    {
+        static const GLenum glsf[]{ GL_NEVER, GL_LESS, GL_EQUAL, GL_LEQUAL, GL_GREATER, GL_NOTEQUAL, GL_GEQUAL, GL_ALWAYS };
+        return glsf[sf];
+    }
+    static inline GLenum getGLlogicOp(asset::E_LOGIC_OP lo)
+    {
+        static const GLenum gllo[]{ GL_CLEAR, GL_AND, GL_AND_REVERSE, GL_COPY, GL_AND_INVERTED, GL_NOOP, GL_XOR, GL_OR, GL_NOR, GL_EQUIV, GL_INVERT, GL_OR_REVERSE,
+            GL_COPY_INVERTED, GL_OR_INVERTED, GL_NAND, GL_SET
+        };
+        return gllo[lo];
+    }
+    static inline GLenum getGLblendFunc(asset::E_BLEND_FACTOR bf)
+    {
+        static const GLenum glbf[]{ GL_ZERO , GL_ONE, GL_SRC_COLOR, GL_ONE_MINUS_SRC_COLOR, GL_DST_COLOR, GL_ONE_MINUS_DST_COLOR, GL_SRC_ALPHA,
+            GL_ONE_MINUS_SRC_ALPHA, GL_DST_ALPHA, GL_ONE_MINUS_DST_ALPHA, GL_CONSTANT_COLOR, GL_ONE_MINUS_CONSTANT_COLOR, GL_CONSTANT_ALPHA, GL_ONE_MINUS_CONSTANT_ALPHA,
+            GL_SRC_ALPHA_SATURATE, GL_SRC1_COLOR, GL_ONE_MINUS_SRC1_COLOR, GL_SRC1_ALPHA, GL_ONE_MINUS_SRC1_ALPHA
+        };
+        return glbf[bf];
+    }
+    static inline GLenum getGLblendEq(asset::E_BLEND_OP bo)
+    {
+        GLenum glbo[]{ GL_FUNC_ADD, GL_FUNC_SUBTRACT, GL_FUNC_REVERSE_SUBTRACT, GL_MIN, GL_MAX };
+        if (bo >= std::extent<decltype(glbo)>::value)
+            return GL_INVALID_ENUM;
+        return glbo[bo];
+    }
```

Comment:
just wondering if it wouldnt be better to make the arrays `constexpr` instead of `const` ?

Reply:
whatever, tbh i think theres no difference between const and constexpr in case of non-func things initialized with literals